kohvis/src/mapping_gd.rs

```diff
@@ -37,20 +38,39 @@ impl Mapping {
             kohonen_path: "".to_string(),
             layers,
             layout_columns: None,
+            kohonen_node: None,
             scale: None,
             colors: ColorPalette::default(),
             font,
         }
     }
 
     #[export]
-    fn _ready(&mut self, _owner: Control) {}
+    fn _ready(&mut self, owner: Control) {
+        self.kohonen_node = util::get_kohonen_node(owner, &self.kohonen_path);
+    }
 
     #[export]
     fn _process(&mut self, owner: Control, _delta: f64) {
         self.update(owner);
     }
 
+    fn update(&mut self, owner: Control) {
+        util::with_kohonen(
+            owner.clone(),
```

Comment:
Isn't `gdnative::Control: Copy`?

Reply:
This should also imply that the `as_ref` dance below should be unnecessary, meaning that `self.kohonen_node.unwrap()` should suffice.